Use secureRender to fix CSP problems with new magento

superdav42 opened the pull request:
Fixes #59 
Unusable in newer Magento without these changes. 
Using the nonce in tracker.js in only needed if the matomo is running on the same domain, which I am but probably isn't so common.

superdav42 committed:
Use secureRender to fix CSP problems with new magento

ihor-sviziev commented on view/frontend/templates/matomo.phtml:
```diff
@@ -29,35 +31,39 @@
 ]); ?>
 </script>
 <?php
+// Render tage without echo so it's added to the dynamic CSP.
```
```suggestion
// Render tag without echo so it's added to the dynamic CSP.
```

superdav42 committed:
Update view/frontend/templates/matomo.phtml

Co-authored-by: Ihor Sviziev <ihor-sviziev@users.noreply.github.com>

ihor-sviziev commented:
@fnogatz could you please review this pull request?

superdav42 committed:
feat: PHP 8.4 / Magento 2.4.8 compatibility via Rector

Modernise codebase using Rector (PHP_80 through PHP_84 set chain + Magento
custom rules) so the module loads cleanly in Magento 2.4.8-p5 on PHP 8.4.

Changes (24 files):
- Explicit nullable parameter types (?Type instead of Type) - PHP 8.4
- Constructor property promotion - PHP 8.0
- readonly properties where applicable - PHP 8.1
- #[\Override] attribute on overridden methods - PHP 8.3
- Match expressions and str_contains/str_starts_with modernisations

Tested by setup:di:compile in goodsalt2 against Magento 2.4.8-p5.

superdav42 committed:
fix: pass Matomo tracker action name
